drivers/misc/bh_fwtable/bh_fwtable.c

```diff
@@ -45,22 +50,65 @@ struct bh_fwtable_data {
 	FwTable fw_table;
 	FlashInfoTable flash_info_table;
 	ReadOnly read_only_table;
+	bool initialized; /* Track if tables have been loaded */
 };
 
 /* Getter function that returns a const pointer to the fw table */
 const FwTable *tt_bh_fwtable_get_fw_table(const struct device *dev)
 {
 	struct bh_fwtable_data *data = dev->data;
 
+	/* Load tables on first access */
+	if (!data->initialized) {
+		if (tt_bh_fwtable_load_tables(dev) != 0) {
+			return NULL;
+		}
+	}
```

Comment:
This is a little bit racey. It should be OK if you want to synchronize with a semaphore though.

To reduce duplication, I would suggest putting this chunk directly inside `tt_bh_fwtable_load_tables()`, and then each function can simply call `tt_bh_fwtable_load_tables()` without any external checks.
```suggestion
	SYS_SEM_LOCK(&data->lock) {
		/* Load tables on first access */
		if (!data->initialized) {
			if (tt_bh_fwtable_load_tables(dev) != 0) {
				return NULL;
			}
		}
	}
```

Reply:
After the above logic has been added to `tt_bh_fwtable_load_tables()`, this section is a bit simpler.
```suggestion
	if(tt_bh_fwtable_load_tables(dev) != 0) {
		return NULL;
	}
```